aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -384,24 +400,34 @@ Tensor sparse_compressed_to_dense(
     dst_transposed.add_(to_transposed_csr);
     return dst_transposed.transpose(batch_ndim, batch_ndim + 1);
   }
-  if (self.layout() == kSparseBsr) {
-    auto crow_indices = self.crow_indices();
-    auto col_indices = self.col_indices();
+  if (self.layout() == kSparseBsr || self.layout() == kSparseBsc) {
+    Tensor compressed_indices;
+    Tensor plain_indices;
+    std::tie(compressed_indices, plain_indices) =
+        AT_DISPATCH_ROW_SPARSE_COMPRESSED_LAYOUTS(
+            self.layout(),
+            "sparse_compressed_to_dense",
+            [&]() {
+              return std::make_pair(self.crow_indices(), self.col_indices());
+            },
+            [&]() {
+              return std::make_pair(self.ccol_indices(), self.row_indices());
+            });
     auto values = self.values();
     Tensor dense = at::zeros(self.sizes(), self.options().layout(kStrided));
     if (self.dim() == 2) {
       // Pad shape so we can treat 2-d like batched, we will squeeze out the
       // phantom batch dim at the end
-      crow_indices = crow_indices.unsqueeze(0);
-      col_indices = col_indices.unsqueeze(0);
+      compressed_indices = compressed_indices.unsqueeze(0);
+      plain_indices = plain_indices.unsqueeze(0);
       values = values.unsqueeze(0);
       dense = dense.unsqueeze(0);
     }
     if (self.dim() > 3) {
       // Flatten batch dims
       auto n_batch_dim = self.dim() - 2;
-      crow_indices = crow_indices.flatten(0, n_batch_dim - 1);
-      col_indices = col_indices.flatten(0, n_batch_dim - 1);
+      compressed_indices = compressed_indices.flatten(0, n_batch_dim - 1);
+      plain_indices = plain_indices.flatten(0, n_batch_dim - 1);
```

Comment:
cc @bhosmer , do we want this op to be performant? If contiguity restriction is lifted, `flatten` will create a new storage.